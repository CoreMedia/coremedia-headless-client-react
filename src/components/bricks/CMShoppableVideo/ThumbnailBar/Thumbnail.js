// @flow
import React from 'react';
import PropTypes from 'prop-types';

import LinkButton from './LinkButton';
import Picture from '../../CMPicture';

type Props = {
  ariaLabel?: string,
  pictureLink: string,
  pictureTitle: string,
  pictureAlt: string,
  handleClick: (ev: SyntheticEvent<HTMLButtonElement>) => void,
  active: boolean,
};

class Thumbnail extends React.Component<Props> {
  static propTypes = {
    ariaLabel: PropTypes.string,
    pictureLink: PropTypes.string.isRequired,
    pictureTitle: PropTypes.string.isRequired,
    pictureAlt: PropTypes.string.isRequired,
    handleClick: PropTypes.func.isRequired,
    active: PropTypes.bool.isRequired,
  };

  shouldComponentUpdate(nextProps: Props) {
    return (
      nextProps.ariaLabel !== this.props.ariaLabel ||
      nextProps.pictureLink !== this.props.pictureLink ||
      nextProps.pictureTitle !== this.props.pictureTitle ||
      nextProps.pictureAlt !== this.props.pictureAlt ||
      nextProps.active !== this.props.active
    );
  }

  render() {
    const { ariaLabel, pictureLink, pictureTitle, pictureAlt, handleClick, active } = this.props;
    return (
      <LinkButton active={active} ariaLabel={ariaLabel} onClick={handleClick}>
        <Picture link={pictureLink} ratio="thumbnail" title={pictureTitle} alt={pictureAlt} />
      </LinkButton>
    );
  }
}

export default Thumbnail;
