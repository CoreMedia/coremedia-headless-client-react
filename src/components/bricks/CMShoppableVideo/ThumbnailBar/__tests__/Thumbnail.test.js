// @flow
import React from 'react';

import Thumbnail from '../Thumbnail';

describe('Thumbnail Component', () => {
  it('should render correctly with default props', () => {
    const handleClick = jest.fn().mockName('handleClick');
    const wrapper = shallow(
      <Thumbnail
        active={false}
        pictureLink="coremedia:///image/6188/data"
        pictureTitle="Title"
        pictureAlt="Alt"
        handleClick={handleClick}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should prevent update', () => {
    const handleClick = jest.fn().mockName('handleClick');
    const props = {
      active: false,
      pictureLink: 'coremedia:///image/6188/data',
      pictureTitle: 'Title',
      pictureAlt: 'Alt',
      handleClick,
    };
    const wrapper = shallow(<Thumbnail {...props} />);
    const value = wrapper.instance().shouldComponentUpdate(props);
    expect(value).toBe(false);
  });
  it('should permit update', () => {
    const handleClick = jest.fn().mockName('handleClick');
    const props = {
      active: false,
      pictureLink: 'coremedia:///image/6188/data',
      pictureTitle: 'Title',
      pictureAlt: 'Alt',
      handleClick,
    };
    const wrapper = shallow(<Thumbnail {...props} />);
    const instance = wrapper.instance();
    const valueA = wrapper.instance().shouldComponentUpdate({ ...props, active: true });
    const valueB = instance.shouldComponentUpdate({
      ...props,
      pictureLink: 'coremedia:///image/6162/data',
    });
    const valueC = instance.shouldComponentUpdate({ ...props, pictureTitle: 'new Title' });
    const valueD = instance.shouldComponentUpdate({ ...props, pictureAlt: 'new Alt' });
    const valueE = instance.shouldComponentUpdate({ ...props, ariaLabel: 'new Label' });
    expect(valueA).toBe(true);
    expect(valueB).toBe(true);
    expect(valueC).toBe(true);
    expect(valueD).toBe(true);
    expect(valueE).toBe(true);
  });
});
