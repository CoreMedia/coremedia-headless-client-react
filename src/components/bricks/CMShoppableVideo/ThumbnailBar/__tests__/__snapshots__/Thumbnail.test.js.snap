// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail Component should render correctly with default props 1`] = `
<LinkButton
  active={false}
  onClick={[MockFunction handleClick]}
>
  <WithTheme(Picture)
    alt="Alt"
    link="coremedia:///image/6188/data"
    ratio="thumbnail"
    theme={
      Object {
        "breakpoints": Object {
          "desktop": 992,
          "large": 1280,
          "small": 480,
          "tablet": 768,
        },
      }
    }
    title="Title"
  />
</LinkButton>
`;
